src/main/scala/replpp/UsingDirectives.scala

```diff
@@ -1,20 +1,32 @@
 package replpp
 
-import java.util.stream.Collectors
-import scala.jdk.CollectionConverters._
+import java.nio.file.Path
+import scala.collection.mutable
 
 object UsingDirectives {
-  val LibDirective = "//> using lib "
+  val Prefix = "//> using"
+  val LibDirective = s"$Prefix lib "
+  val FileDirective = s"$Prefix file "
 
-  def findDeclaredDependencies(source: String): Seq[String] = {
-    source
-      .lines()
+  def findImportedFilesRecursively(lines: IterableOnce[String], visited: Set[os.Path] = Set.empty): Set[os.Path] = {
+    val files = scanFor(FileDirective, lines).iterator.map { pathStr =>
+      if (Path.of(pathStr).isAbsolute) os.Path(pathStr)
+      else os.pwd / os.RelPath(pathStr)
```

Comment:
I think it would be way more convenient if paths were interpreted relative to the file that's doing the importing instead of the current working dir.

Reply:
true! yes that makes way more sense, thanks!

Reply:
that should probably be the way for predefFiles, scriptFile and anything down from there...
in other words: only `--predefCode` should be looking relative from the current working dir